infra/terraform/README.md

```diff
@@ -21,35 +21,60 @@ prod
 
 By default, the selected region is `us-east-1` and can be configured by editing the TF variables used for applying the TF plan, e.g. the one in [`./hash/prod-usea1.tfvars`](./hash/prod-usea1.tfvars).
 
-# Deploying
+# Naming convention
 
-Deployment currently relies on a couple of manual steps - but is to be automated in the future. The container registries need to have images pushed in the correct place for the applications to start and the database has to be migrated manually. The order of executions for deployment:
+Resources use the following naming convention:
 
-1.  Deploy infrastructure with terraform
-1.  Migrate databases
-1.  Build/push docker images
+```text
+{OWNER}-{PROJECT}-{ENV}-{REGION}-{RESOURCE DESCRIPTION}
 
-## Deploy infrastructure with terraform
+OWNER : 'h' for hash
+PROJECT : 'hash'
+ENV : 'dev' or 'prod'  (maybe others too later)
+REGION : AWS region shortened -- 'usea1' etc.
+RESOURCE DESCRIPTION : a short description of the resource e.g. 'subnetpub1'
+```
 
-You must prepare a `tfvars` file containing secrets to spin up the environment. This can be done by creating a file in `./hash/prod.secrets.tfvars` containing the following definitions (change the values):
+Fields may _not_ include any hyphens. Example names: `h-hash-prod-usea1-vpc`, `h-hash-dev-usea2-apisvc`.
 
-```tfvars
-hash_seed_users = [
-  { "email" = "admin@hash.ai", "shortname" = "instance-admin", "preferredName" = "Instance Admin", "password" = "changeme", "isInstanceAdmin" = true }
-]
+(Inspired by https://stepan.wtf/cloud-naming-convention/)
 
-hash_system_user_password = "changeme"
+The region and environment fields are set by the variables supplied in the `*.tfvars` file. The config automatically shortens AWS region names, for example, `us-east-1` will be converted to `usea1`.
 
-hash_block_protocol_api_key = "Generate at https://blockprotocol.org/settings/api-keys"
+# Deploying
 
-kratos_secrets_cookie = "VERY-INSECURE-AND-SHOULD-ONLY-BE-USED-IN-DEV"
-kratos_secrets_cipher = "32-LONG-SECRET-NOT-SECURE-AT-ALL"
+Deployment currently relies on a couple of manual steps - but is to be automated in the future. The container registries need to have images pushed in the correct place for the applications to start and the database has to be migrated manually. The order of executions for deployment:
 
-kratos_api_key = "secret"
+1.  Deploy infrastructure with terraform
+1.  Migrate databases
+1.  Build/push docker images
+
+## Deploy infrastructure with terraform
 
-pg_superuser_password    = "changeme"
-pg_kratos_user_password  = { raw = "changeme", hash = "See guide below to generate this" }
-pg_graph_user_password   = { raw = "changeme", hash = "See guide below to generate this" }
+Secret environment should be provided in HashiCorp Vault. These are expected in a kvv2 path starting with `pipelines/hash/` and ending with the environment name, e.g. `automation/pipelines/hash/prod`. The following secrets are expected:
```

Comment:
where does `automation` come from here?

Reply:
That's the engine name, I've tried to make it more clear in 82fe967394043693d34176331d985bef0c8d92ce